Tools/Scripts/webkitpy/w3c/test_exporter.py

```diff
@@ -240,13 +240,28 @@ def create_branch_with_patch(self, patch):
             self._run_wpt_git(['checkout', '-b', self._branch_name])
 
         try:
-            self._run_wpt_git(['apply', '--index', patch, '-3'])
+            output = self._run_wpt_git(['apply', '--index', patch, '-3'], capture_output=True)
+            if output.returncode:
+                _log.error(f'Failed to apply patch!')
+                _log.error(f"{output.stdout.decode('utf-8')}\n{output.stderr.decode('utf-8')}")
+                return False
         except Exception as e:
             _log.warning(e)
             return False
-        if self._run_wpt_git(['commit', '-a', '-m', self._commit_message]).returncode:
-            _log.error('No changes to commit! Exiting...')
+
+        # Check if there are no changes to commit
+        if self._run_wpt_git(['add', '--all']).returncode:
+            _log.error('Failed to add changes')
+            return False
+
+        if not self._run_wpt_git(['diff', '--cached', '--quiet']).returncode:
+            _log.warning('No changes to upstream detected! Cancelling export.')
+            self.delete_local_branch(is_success=True)
+            sys.exit(0)
```

Comment:
Why `sys.exit(0)` and not `return False` like all the above?

Reply:
I feel like this could be cleaned up in the future, along with the rest of the overall code flow, but I guess it's fine for now.